vm/array.go

```diff
@@ -46,6 +46,32 @@ func builtinArrayInstanceMethods() []*BuiltinMethodObject {
 			// a[-1] # => "c"
 			// a[-3] # => "a"
 			// a[-7] # => nil
+			//
+			// # Double indexing, second argument specifies the count of the elements
+			// a[1, 3]  # => [2, 3, "a"]
+			// a[1, 0]  # => [] <-- Zero count is empty
+			// a[1, 5]  # => [2, 3, "a", "b", "c"]
+			// a[1, 10] # => [2, 3, "a", "b", "c"]
+			// a[-3, 2] # => ["a", "b"]
+			// a[-3, 5] # => ["a", "b", "c"]
+			// a[5, 1]  # => ["c"]
+			// a[6, 1]  # => []
+			// a[7, 1]  # => nil
+			//
+			// Special case 1:
+			// a[6]    # => nil
+			// a[6, 1] # => []  <-- Not nil!
+			// a[7, 1] # => nil <-- Because it is really out of the edge of the array
+			//
+			// Special case 2: Second argument is negative
```

Comment:
According to the tests, I thought the second argument can't be negative?

Reply:
According to `irb`:
```
╰─⠠⠵ irb
2.4.1 :001 > a = [1,2,3,4,5]
 => [1, 2, 3, 4, 5] 
2.4.1 :002 > a[1, -1]
 => nil 
2.4.1 :003 > a[6, -1]
 => nil 
2.4.1 :004 > a[-3, -1]
 => nil 
2.4.1 :005 > a[-8, -1]
 => nil 
2.4.1 :006 > 
```

Reply:
In Goby, should we always raise error when encounter the second argument negative case?

Reply:
I think we should. Will the result always be `nil` when the second argument is negative?

Reply:
Okay, let's change the behaviour when inputing the negative value in second argument! => It should raise `ArgumentError` instead of `nil`